lib/hls/hls_parser.js

```diff
@@ -5011,6 +5017,19 @@ shaka.hls.HlsParser.EXTENSION_MAP_BY_CONTENT_TYPE_ = {
 };
 
 
+/**
+ * MIME types without init segment.
+ *
+ * @const {!Array.<string>}
+ * @private
+ */
+shaka.hls.HlsParser.MIME_TYPES_WITHOUT_INIT_SEGMENT_ = [
```

Comment:
You should construct a Set from this array, so we don't have to do an array lookup for every segment.

Reply:
Done!